src/redux/books/booksSlice.jsx

```diff
@@ -1,27 +1,45 @@
-import { createSlice } from "@reduxjs/toolkit";
+import axios from 'axios';
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
 import { v4 as uuidv4 } from "uuid";
 
-const initialState = {
-  books: [
-    {
-      "id": uuidv4(),
-      "title": "The Great Gatsby",
-      "author": "John Smith",
-      "category": "Fiction"
-    },
-    {
-      "id": uuidv4(),
-      "title": "Anna Karenina",
-      "author": "Leo Tolstoy",
-      "category": "Fiction"
-    },
-    {
-      "id": uuidv4(),
-      "title": "The Selfish Gene",
-      "author": "Richard Dawkins",
-      "category": "Nonfiction"
+const APPID = '1ZndMyOgnRMUYmlQAdZA';
+// const APPID = 'LQYGPUQgAZCQeGfRpgA8';
+const url = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${APPID}/books`;
+
+const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get(`${url}`);
+  console.log('book response', response.data)
+  return response.data;
+});
+
+const addTheBook = createAsyncThunk('books/addTheBook', async (book, API) => {
+  try {
+    await axios.post(`${url}`, book);
+    API.dispatch(fetchBooks());
+    const response = API.getState().books;
+    return response;
+  } catch (error) {
+    throw new Error(error);
+  }
+});
```

Comment:
- [ ] Please consider changing the books' state based on this action instead of fetching the books when you perform a change to the DB. I would suggest that instead of doing the `API.dispatch(fetchBooks());` inside this thunk, you push the newly created book into the books array inside your `extraReducers` fulfilled case, like this:

```suggestion
const addTheBook = createAsyncThunk('books/addTheBook', async (book) => {
  try {
    const response = await axios.post(`${url}`, book);


    return response.data;
  } catch (error) {
    throw new Error(error);
  }
});
```

```js
  extraReducers: (builder) => {
...
    builder.addCase(addTheBook.fulfilled, (state, action) => {
      state.loading = false;
      state.books.push(action.payload);
    })
...
}
```
This way you avoid making unnecessary calls to the server and improve a lot your app performance! 🤓 🔧

Reply:
@anagudelogu I applied your suggestion, but it doesn't work. It shows an error.
![state error](https://github.com/Aqurds/bookstore/assets/39187898/a60bbb96-867e-4aa2-9fcd-95f7e6571fe9)